File: app/src/main/java/com/islam/basepropject/ui/ExFetchData/Fragment1.java
package com.islam.basepropject.ui.ExFetchData;


import android.os.Bundle;
import android.util.Log;
import android.view.View;

import androidx.fragment.app.Fragment;

import com.google.gson.JsonElement;
import com.islam.basepropject.R;
import com.islam.basepropject.data.Repository;
import com.islam.basepropject.project_base.base.fragments.BaseFragment;
import com.islam.basepropject.project_base.base.other.BaseViewModel;
import com.islam.basepropject.project_base.utils.FragmentManagerUtil;
import com.islam.basepropject.project_base.utils.network.RetrofitObserver;
import com.islam.basepropject.project_base.views.OnViewStatusChange;
import com.islam.basepropject.ui.ExRecyclerView.Fragment3;

import io.reactivex.disposables.Disposable;
import io.reactivex.observers.DisposableSingleObserver;

/**
 * A simple {@link Fragment} subclass.
 */
public class Fragment1 extends BaseFragment<Fragment1.ViewModel> {


    public Fragment1() {
        // Required empty public constructor
    }

    @Override
    protected void onLaunch() {
        initContentView(R.layout.fragment_fragment1);
        initToolbar(R.string.title1, false);
        initViewModel(getActivity(), ViewModel.class);
    }

    @Override
    protected void onViewCreated(View view, ViewModel viewModel, Bundle instance) {
        markScreenAsCompleted();
        view.findViewById(R.id.btn_fetch).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                getViewModel().loadProviders(view.findViewById(R.id.btn_fetch));
            }
        });
        // loadData();
        (view).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                FragmentManagerUtil.replaceFragment(getFragmentManager(), new Fragment3(), true);
            }
        });

    }


    @Override
    protected void setUpObservers() {

    }

    public static class ViewModel extends BaseViewModel {

        public void loadProviders(OnViewStatusChange onViewStatusChange) {
            Repository repository = new Repository();
            repository.getProvidresList()
                    .subscribeOn(getSchedulerProvider().io())
                    .observeOn(getSchedulerProvider().ui())
                    .subscribeWith(new RetrofitObserver<JsonElement>(this, onViewStatusChange) {
                        @Override
                        public void onResultSuccess(JsonElement o) {
                            Log.i("network", o.toString());
                        }
                    });

//            addDisposable();

        }
    }
}

File: app/src/main/java/com/islam/basepropject/ui/ExFetchData/Fragment1.kt
package com.islam.basepropject.ui.ExFetchData


import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment
import androidx.lifecycle.viewModelScope
import com.islam.basepropject.R
import com.islam.basepropject.data.Repository
import com.islam.basepropject.pojo.Article
import com.islam.basepropject.project_base.POJO.ApiResponse
import com.islam.basepropject.project_base.base.fragments.BaseFragment
import com.islam.basepropject.project_base.base.other.BaseViewModel
import com.islam.basepropject.project_base.base.other.network.Result
import com.islam.basepropject.project_base.views.OnViewStatusChange
import com.islam.basepropject.ui.ExDialogs.Fragment5
import kotlinx.android.synthetic.main.fragment_fragment1.*
import kotlinx.coroutines.launch


/**
 * A simple [Fragment] subclass.
 */
class Fragment1 : BaseFragment<ViewModel>() {
    override var fragmentTag = "Fragment1"

    override fun onLaunch() {
        initContentView(R.layout.fragment_fragment1)
        initToolbar(R.string.title1)
        initViewModel(this, ViewModel::class.java)
    }

    override fun onViewCreated(view: View, viewModel: ViewModel?, instance: Bundle?) {

        btnFetch.setOnClickListener { viewModel!!.loadProviders(btnFetch.id) }
        addSensitiveInputs(editText)
        view.setOnClickListener { navigate(Fragment5(), addToBackStack = true) }
    }


    override fun setUpObservers() {}
}






